### PR TITLE
Karma support

### PR DESCRIPTION
**Pull request by rjatkins**
Implements #188

### CODE REVIEW

File: src/special/karma.js [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+  return obj;
+}
+
+function collectFrameworks(config, karmaPluginsInstalled) {
+  // karma-x plugins define frameworks using the structure { 'framework:<name>': ['factory', f] }
+  // generate a lookup map { '<name>': 'karma-x', ... }
+  const frameworkMapping = {};
+  // skip plugins with more specific names, such as launchers and reporters
+  karmaPluginsInstalled.filter(plugin => !plugin.endsWith('-launcher') && !plugin.endsWith('-reporter')).forEach((genericPlugin) => {
+    const p = tryRequire(genericPlugin);
```
Comment (rjatkins) [inline src/special/karma.js]: This specific approach - using node's require to try to pull in the relevant module - seems to be against the philosophy of the rest of the project. Sadly the information we need to resolve frameworks/browsers/reporters/etc to plugins isn't quite available statically, which means we need to find the default module in each plugin and parse out the map from its module.export. Any guidance on how to do that cheaply would be much appreciated!

**Comment by rumpl** [comment]
The build fails on node 6, I'm not against dropping node 6 support...

File: src/special/karma.js (hunk 1 above)
Comment (rjatkins) [reply src/special/karma.js]: Opted to replace it with parser code, so I can guarantee it works in my own project.

**Comment by rjatkins** [comment]
> The build fails on node 6, I'm not against dropping node 6 support...

It's easy enough to stop using `Object.values` and make it backwards compatible again, though I'm really confused why `@babel/polyfill` isn't triggering in appveyor, but works fine in travis. Doing that instead.

**Comment by rumpl** [comment]
🎉

Thanks!

**Pull request merged**